### PR TITLE
fix: Make the tracer work with the new interpreter runtime

### PR DESCRIPTION
**Pull request by daemonsy**
## What's up? 

Context can be on the query on passed directly as denoted by this statement. By without passing a second argument to `fetch`, it raises so we never get to the `data.fetch(:query).context` branch.

### CODE REVIEW

File: spec/apollo-federation/tracing_spec.rb [hunk 1]
```diff
@@ -4,307 +4,320 @@
 require 'apollo-federation'
 
 RSpec.describe ApolloFederation::Tracing do
-  let(:base_schema) do
-    Class.new(GraphQL::Schema) do
-      use ApolloFederation::Tracing
+  RSpec.shared_examples 'a basic tracer' do
```
Comment (daemonsy) [inline spec/apollo-federation/tracing_spec.rb]: Large diff to the spec mostly due to indentation and an edit on one of the line. I'll highlight where the change is

File: spec/apollo-federation/tracing_spec.rb [hunk 2]
```diff
@@ -4,307 +4,320 @@
...
+                  }, {
+                    response_name: 'name',
+                    type: 'String!',
+                    start_time: 9,
+                    end_time: 10,
+                    parent_type: 'Item',
+                    error: [{
+                      message: "Can't continue with this query",
+                      location: [{ line: 1, column: 15 }],
+                      json: %({"message":"Can't continue with this query","locations":[{"line":1,"column":15}],"path":["items",1,"name"]}),
```
Comment (daemonsy) [inline spec/apollo-federation/tracing_spec.rb]: I removed some whitespace here. Previously it had some whitespace added into it.

File: spec/apollo-federation/tracing_spec.rb [hunk 3]
```diff
@@ -4,307 +4,319 @@
...
-      )
+  context 'with the new interpreter' do
+    let(:base_schema) do
+      Class.new(GraphQL::Schema) do
+        use ApolloFederation::Tracing
+        use GraphQL::Execution::Interpreter
+      end
     end
+
+    it_behaves_like 'a basic tracer'
```
Comment (daemonsy) [inline spec/apollo-federation/tracing_spec.rb]: New test case using the interpreter

File: lib/apollo-federation/tracing/tracer.rb [hunk 4]
```diff
@@ -116,16 +116,18 @@ def self.execute_field(data, &block)
...
           field_type = context.field.type.to_s
           parent_type = context.parent_type.graphql_name
-        else # legacy runtime
+        else # interpreter runtime
           path = data.fetch(:path)
-          field_name = data.fetch(:field).graphql_name
-          field_type = data.fetch(:field).type.unwrap.graphql_name
+          field = data.fetch(:field)
+          field_name = field.graphql_name
+          field_type = field.type.unwrap.graphql_name
```
Comment (daemonsy) [inline lib/apollo-federation/tracing/tracer.rb]: Good call to write a spec @grxy . This has always been broken by the test caught it. The `graphql_name` of the type doesn't include whether it's nullable, hence the need to check the field

File: lib/apollo-federation/tracing/tracer.rb [hunk 5]
```diff
@@ -116,16 +116,18 @@ def self.execute_field(data, &block)
...
 
-        # interpreter runtime
+        # legacy runtime
         if data.include?(:context)
           path = context.path
           field_name = context.field.graphql_name
           field_type = context.field.type.to_s
           parent_type = context.parent_type.graphql_name
-        else # legacy runtime
+        else # interpreter runtime
```
Comment (daemonsy) [inline lib/apollo-federation/tracing/tracer.rb]: I believe we got the legacy and new interpreter the wrong way round. I get into this branch

**Comment by daemonsy** [comment]
Hey @noaelad :wave: this is a small fix before we can switch our own stuff to use the new interpreter.

File: spec/apollo-federation/tracing_spec.rb (hunk 3 above)
Comment (grxy) [reply spec/apollo-federation/tracing_spec.rb]: Should we also test with the legacy runtime?

Comment (daemonsy) [reply spec/apollo-federation/tracing_spec.rb]: I added another test case. Good catch, I was under the impression that the examples run as a suite, but that doesn't make sense. 

Also, the `base_schema` variable wasn't been overwritten correctly. I removed it from the example so the actual test cases have to define it correctly.

File: lib/apollo-federation/tracing/tracer.rb [hunk 6]
```diff
@@ -116,16 +116,17 @@ def self.execute_field(data, &block)
...
           path = data.fetch(:path)
-          field_name = data.fetch(:field).graphql_name
-          field_type = data.fetch(:field).type.unwrap.graphql_name
+          field = data.fetch(:field)
+          field_name = field.graphql_name
+          field_type = field.type.to_type_signature
```
Comment (daemonsy) [inline lib/apollo-federation/tracing/tracer.rb]: This seems like it's what we want. The other hacky way was leaky

**Pull request merged**